packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionsTable/TransactionsTableCctp.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useEffect, useMemo, useState } from 'react'
+import { useAccount } from 'wagmi'
+
+import { TransactionsTableDepositRow } from './TransactionsTableDepositRow'
+import { TransactionsTableWithdrawalRow } from './TransactionsTableWithdrawalRow'
+import { isDeposit, isWithdrawal } from '../../../state/app/utils'
+import { TableBodyLoading } from './TableBodyLoading'
+import { TableBodyError } from './TableBodyError'
+import { TableActionHeader } from './TableActionHeader'
+import {
+  EmptyTableRow,
+  PageParams,
+  TableStatus,
+  TransactionsTableHeader
+} from './TransactionsTable'
+import { TransactionsTableSwitch } from './TransactionsTableSwitch'
+import { useCctpFetching, useCctpState } from '../../../state/cctpState'
+import { useNetworksAndSigners } from '../../../hooks/useNetworksAndSigners'
+import { MergedTransaction } from '../../../state/app/state'
+import { useAccountType } from '../../../hooks/useAccountType'
+
+export function TransactionsTableCctp() {
+  const { address } = useAccount()
+  const [type, setType] = useState<'deposits' | 'withdrawals'>('deposits')
+  const { isSmartContractWallet } = useAccountType()
+  const [pageParams, setPageParams] = useState<PageParams>({
+    searchString: '',
+    pageNumber: 0,
+    pageSize: 10
+  })
+  const { l1 } = useNetworksAndSigners()
+  const { transfers, depositIds, withdrawalIds } = useCctpState()
+  const {
+    depositsError,
+    withdrawalsError,
+    isLoadingDeposits,
+    isLoadingWithdrawals
+  } = useCctpFetching({
+    l1ChainId: l1.network.id,
+    walletAddress: address,
+    pageSize: pageParams.pageSize,
+    pageNumber: pageParams.pageNumber,
+    type
+  })
+  const isLoading =
+    type === 'deposits' ? isLoadingDeposits : isLoadingWithdrawals
+  const hasError = type === 'deposits' ? depositsError : withdrawalsError
+
+  const status = useMemo(() => {
+    if (isLoading) return TableStatus.LOADING
+    if (hasError) return TableStatus.ERROR
+    return TableStatus.SUCCESS
+  }, [hasError, isLoading])
+
+  const transactions = useMemo(() => {
+    const startIndex = pageParams.pageNumber * pageParams.pageSize
+    const ids = type === 'deposits' ? depositIds : withdrawalIds
+    return ids
+      .map(id => transfers[id])
+      .slice(
+        startIndex,
+        startIndex + pageParams.pageSize
+      ) as unknown as MergedTransaction[]
+  }, [
+    depositIds,
+    pageParams.pageNumber,
+    pageParams.pageSize,
+    transfers,
+    type,
+    withdrawalIds
+  ])
+
+  useEffect(() => {
+    setPageParams(prevPageParams => ({
+      ...prevPageParams,
+      pageNumber: 0
+    }))
+  }, [setPageParams, type])
+
+  const tabs = useMemo(() => {
+    return [
+      {
+        handleClick: () => setType('deposits'),
+        isActive: type === 'deposits',
+        text: 'Deposits'
+      },
+      {
+        handleClick: () => setType('withdrawals'),
+        isActive: type === 'withdrawals',
+        text: 'Withdrawals'
```

Comment:
Thinking we should call these `To Mainnet` and `To Arbitrum One`, similar to how we have for the regular transfers. We can use the `getNetworkName` util.